### PR TITLE
docs(readme): Document most configuration options

### LINKED ISSUE #29: docs: document the different configuration options
in the README or alternatively in a dedicated documentation file. Specifically requested for the `schema` configuration (as an example) [here](https://github.com/Optravis-LLC/jooq-gradle/issues/26#issuecomment-2222404997).

as a starter, until we get to it, my config from another project:

```
@OptIn(ExperimentalJooqGeneratorConfig::class)
jooqGenerator {
    val dbUserName = props.getProperty("spring.datasource.hikari.username")
    val dbPassword = props.getProperty("spring.datasource.hikari.password")
    containerConfig = ContainerConfig(
        image = "postgres:15.4",
        port = 5432,
        environment = mapOf(
            "POSTGRES_DB" to props.getProperty("db.name"),
            "POSTGRES_USER" to dbUserName,
            "POSTGRES_PASSWORD" to dbPassword,
        ),
    )
    jooqDbConfig = JooqDatabaseConfig.postgres(recordVersionFields = listOf("version"))
    generatorConfig = JooqGeneratorConfig(
        generatorType = GeneratorType.Java,
        deprecateUnknownTypes = true,
        javaTimeTypes = false,
        daos = false,
        pojos = false,
    )
    connectionConfig = DbConnectionConfig(
        user = dbUserName,
        password = dbPassword,
        urlTemplate = "jdbc:postgresql://localhost:{{port}}/${props.getProperty("db.name")}",
    )
    packageName = "ch.difty.scipamato.core.db"
}
````

### PR DESCRIPTION
**Pull request by jcornaz**
Document most configuration options in the readme.

I intentionally don't document how to configure the database name, user and password, because those should normally not be useful for people using Postgres. Those are ephemeral, and for all the user knows they could be generated on the fly each time the plugin runs.

fix #29

### CODE REVIEW

**Comment by ursjoss** [comment]
Looks very good, thanks. Maybe some aspects to consider and decide upon:

- In the readme, we could state that all configured values in the example are the actual default values used.
- if so, we wouldn't need to mention it for the container version specifically.
- Maybe we also mention the possible configuration options where they are limited.

Not sure if I fully get your statement about database name, user and password in the PR description. But maybe we can quickly discuss it offline.

File: README.md [hunk 1]
```diff
@@ -21,36 +21,60 @@ plugins {
...
 ```kotlin
 import com.optravis.jooq.gradle.ContainerConfig
 import com.optravis.jooq.gradle.ExperimentalJooqGeneratorConfig
+import com.optravis.jooq.gradle.GeneratorType
+import com.optravis.jooq.gradle.JooqDatabaseConfig
 import com.optravis.jooq.gradle.JooqGeneratorConfig
 
+
+// If defined, the `group` determines the package name of the generated code 
+group = "org.jooq.generated"
```
Comment (bartkummel) [inline README.md:46]: I'm nor sure if this is a good example. Generally speaking, the `group` should have something to do with the project. Maybe a better example would be
```
group = "my.project"
```
?

File: README.md [hunk 2]
```diff
@@ -21,36 +21,60 @@ plugins {
...
+    // Configure jOOQ database
+    jooqDbConfig.set(
+        JooqDatabaseConfig.postgres(
+            schema = "public",
+            recordVersionFields = emptyList(),
+        )
+    )
+
+    // Override the package name
+    packageName.set("$group.jooq") // If the group is not set, the default is `org.jooq.generated`
```
Comment (ursjoss) [inline README.md:63]:
```suggestion
    packageName.set("$group.jooq") // If neither the package name nor the group is set explicitly, the package name defaults to `org.jooq.generated`
```

File: README.md (hunk 1 above)
Comment (ursjoss) [reply README.md:46]: I agree

File: README.md [hunk 3]
```diff
@@ -21,36 +21,60 @@ plugins {
...
-
 ```kotlin
 import com.optravis.jooq.gradle.ContainerConfig
 import com.optravis.jooq.gradle.ExperimentalJooqGeneratorConfig
+import com.optravis.jooq.gradle.GeneratorType
+import com.optravis.jooq.gradle.JooqDatabaseConfig
 import com.optravis.jooq.gradle.JooqGeneratorConfig
 
+
+// If defined, the `group` determines the package name of the generated code
```
Comment (ursjoss) [inline README.md:45]:
```suggestion
// The `group` determines the package name of the generated code,
// unless you specify the package name explicitly.
```

File: README.md (hunk 1 above)
Comment (jcornaz) [reply README.md:46]: Yes, that's true. Here my intent was not to suggest what one should write, but rather what is the default if not set.

But after a discussion with @ursjoss, we decided to remove the default value of `org.jooq.generated` as that's misleading. Instead, the task will fail if neither the group, nor the package name is defined.

And I will apply your suggestion for the readme @bartkummel ;-)

File: README.md [hunk 4]
```diff
@@ -25,30 +26,49 @@ jooqGenerator {
...
+* Add the generated jOOQ code to the main source set
 
 > [!NOTE]
 >
 > This plugin requires a `docker` installation
 
 ## Configuration
 
-In general, this gradle plugin aims to require as little configuration as possible.
+In general, this gradle plugin aims to require as little configuration as possible and all configuration is optional.
```
Comment (ursjoss) [inline README.md:37]:
```suggestion
In general, this gradle plugin aims to require as little configuration as possible and almost all configuration is optional.
```

**Review by ursjoss — APPROVED** [review]

**Pull request merged**